pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -616,7 +616,7 @@ func (r *OnlineUpgradeReconciler) promoteReplicaBSubclusters(ctx context.Context
 		return ctrl.Result{}, err
 	}
 	sbPFacts.Invalidate()
-	actor := MakeAlterSubclusterTypeReconciler(r.VRec, r.Log, r.VDB, sbPFacts, r.Dispatcher)
+	actor := MakeAlterSubclusterTypeReconciler(r.VRec, r.Log, r.VDB, sbPFacts, r.Dispatcher, true /* forUpgrade */)
```

Comment:
If you add alter subcluster to sandboxsubcluster_reconiler, do you still need alter here?

Reply:
It should not. When duplicating the main cluster, the subcluster types are recorded into the sandbox subcluster types in moveReplicaGroupBSubclusterToSandbox(). 
So onlineupgrade_reconciler should not alter it again. I will do more testing before removing this.